libcontainer/cgroups/systemd/v2.go

```diff
@@ -292,6 +304,38 @@ func (m *unifiedManager) Apply(pid int) error {
 	if err := fs2.CreateCgroupPath(m.path, m.cgroups); err != nil {
 		return err
 	}
+
+	// When systemd creates a unit owned by a user other than the
+	// user running the service manager (usually root), it chowns
+	// the cgroup directory and the three files mentioned below.
+	// This enables the service to create new scopes/slices and set
+	// the cgroup for its child processes/threads.
+	//
+	// We follow systemd's lead and chown only those same files.
+	goodToChown := map[string]struct{}{
+		"cgroup.procs":           struct{}{},
+		"cgroup.subtree_control": struct{}{},
+		"cgroup.threads":         struct{}{},
+	}
+	if m.uid != nil {
+		// TODO: when go < 1.16 is no longer supported, switch to filepath.WalkDir
+		err := filepath.Walk(m.path, func(path string, info os.FileInfo, err error) error {
```

Comment:
So, are we expecting to have sub-directories here? AFAICS there can be none, or am I missing something?

Reply:
Just the one level at this time.  But I thought it better not to couple tightly to this assumption.  Do you prefer to explicitly chown exactly the 4 files needed at this time (the cgroup dir, and the 3 files within it).

Reply:
I changed it to explicitly mention (and chown) the dir and 3 files needed; no more `Walk`.